### DISCUSSION TITLE
Async testing documentation is missing details on selecting anyio backend

### DISCUSSION THREAD

**Discussion by Insolites** [fastapi/fastapi#11280] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# This is just the async testing example!
@pytest.mark.anyio
async def test_root():
    async with AsyncClient(app=app, base_url="http://test") as ac:
        response = await ac.get("/")
```

### Description

This is due to this change: https://fastapi.tiangolo.com/release-notes/?h=trio#0690

Pytest's AnyIO plugin will try to run tests on all supported backends: https://anyio.readthedocs.io/en/stable/testing.html#specifying-the-backends-to-run-on

Unfortunately if you install FastAPI the usual way and don't explicitly install `trio`, then you don't actually have all the supported backends.

So if you simply follow the [instructions for running async tests](https://fastapi.tiangolo.com/advanced/async-tests/) then you will get a really huge stack trace with:
```
FAILED tests/test_main.py::test_hello[trio] - ModuleNotFoundError: No module named 'trio'
```
A possible solution as described [here](https://github.com/agronholm/anyio/discussions/407) and in the anyio docs is to add a `conftest.py` to select just the `asyncio` backend:
```
import pytest
@pytest.fixture
def anyio_backend():
    return 'asyncio'
```
If there's any way this could be done helpfully/magically by FastAPI that would be great, if not it should be mentioned in the documentation.

### Operating System

Linux

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

Python 3.11.6

**Comment by gioscarda** [comment]
@Insolites I fixed it by changing the decorator:

```
@pytest.mark.parametrize('anyio_backend', ['asyncio'])
async def test_root(anyio_backend):
    ...
```

but for sure the FastAPI docs should be updated.

**Comment by stephanschielke** [comment]
Also, if you want to set this for different scopes (other than on the function level via a decorator):
```
pytestmark = pytest.mark.parametrize("anyio_backend", ["asyncio"], scope="module")
```

**Reply by serhiiur** [reply]
You could also explicitly specify backend for your async tests in <ins>conftest.py</ins>:
```python
import pytest

@pytest.fixture(scope="session")
def anyio_backend() -> str:
    return "asyncio"
```
Also check [Testing with AnyIO](https://anyio.readthedocs.io/en/stable/testing.html) section.
